src/Application/Actions/Donations/Update.php

```diff
@@ -99,7 +100,19 @@ protected function action(): Response
         }
 
         if ($donationData->status === 'Cancelled') {
-            return $this->cancel($donation);
+            try {
+                return $this->cancel($donation);
+            } catch (LockWaitTimeoutException $exception) {
+
+                // we could auto retry, but there is probably no point - the other process is likely to have made the donation non-cancelable.
+
+                $message = "Donation ID {$this->args['donationId']} is locked by another process, could not be set to status {$donationData->status}";
+                $this->logger->warning($message);
```

Comment:
Open question, should this be a warning or an error?

Reply:
I think probably a warning - it's not something we can expect to 100% prevent. Errors should generally be things we can fix.